scripts/ButterParticles.gd

```diff
@@ -10,9 +10,14 @@ extends Particles2D
 func _ready():
 	self.emitting = true
 	self.one_shot = true
-	self.lifetime = 10
-	get_node("ButterParticlesLifetime").wait_time = 15
+	self.lifetime = rand_range(5, 25)
+	get_node("ButterParticlesLifetime").wait_time = rand_range(10, 35)
 
 
 func _on_ButterParticlesLifetime_timeout():
-	queue_free()
+	var n_particles = get_parent().get_children().size()
+	
+	if n_particles <= 150:
```

Comment:
I think that it will be better to create a variable with a maximum particles number

Reply:
👍

Reply:
Something like this maybe?
https://gamedev.stackexchange.com/questions/153968/how-do-i-access-a-variable-from-my-code-in-godots-editor

Reply:
(for tweaking in editor)

Reply:
It might be done that way. We can see that later in the editor. My consideration was not to have unnamed magical value. After some time, we will forget what 150 was :))

Reply:
That's a good call!

Reply:
I'm leaving it as a const for now, like adviced - not exporting to Inspector for now